eva-events/src/main/kotlin/com/razz/eva/events/SharedMutableFlowEventBus.kt

```diff
@@ -0,0 +1,64 @@
+package com.razz.eva.events
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExecutorCoroutineDispatcher
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import mu.KotlinLogging
+import java.io.Closeable
+import java.util.concurrent.Executors
+
+class SharedMutableFlowEventBus(
+    replay: Int,
+    extraBufferCapacity: Int,
+    onBufferOverflow: BufferOverflow,
+    consumers: List<EventConsumer>,
+    val context: ExecutorCoroutineDispatcher = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
+) : EventPublisher, Closeable {
+
+    private val logger = KotlinLogging.logger {}
+    private val consumingJob = Job()
+    private val delegate = MutableSharedFlow<IntegrationModelEvent>(replay, extraBufferCapacity, onBufferOverflow)
+    private val consumerMap: Map<IntegrationModelEvent.EventName, List<EventConsumer>> = consumers
+        .flatMap { consumer -> consumer.eventNames.map { name -> name to consumer } }
+        .groupByTo(mutableMapOf(), { (name, _) -> name }, { (_, consumer) -> consumer })
+
+    fun start() {
+        CoroutineScope(context + consumingJob).launch {
+            delegate.collect { event ->
```

Comment:
it will be cool to process events of different models in parallel, but simple implementation is fine for now

Reply:
I had same thoughts and decided to left opportunity for sophisticated implementations to eva customers.